specs/2019-03/Cloning Branch Configuration/Cloning-Branch-Config-Spec.md

```diff
@@ -0,0 +1,38 @@
+# App Center Build Clone Branch Configuration Proposal
+
+This proposal details the approach the App Center team will take to address a heavily voiced frustration regarding branch configurations. We expect to start work on implementing this feature in the second quarter of 2019.
+
+## Problem Statement
+
+Currently, each time a new branch needs to be configured, one has to configure it completely from scratch. Existing configurations are not available for re-use. This is especially cumbersome for a branch configuration with many environment variables and when dealing with the management of signing credentials. Building a new branch via API today requires full configuration of the branch and uploading secrets.
+
+## Goal
+
+Given that at least one branch is already configured for a specific app, one can clone a configured branch’s configuration and apply it to a non-configured branch.
+
+## Solution
+
+### UI Experience
+* After navigating to a branch configuration page, an icon will be visible to clone a different branch’s configuration and apply it to the current branch.
+* App Center Build will populate existing build configurations to clone.
+* App Center will clone the following from a configured branch’s configuration:
+	* Build settings – Includes toolset versions, build triggers, version number, and launch tests.
+	* Environment variables – Variable names and values will be cloned. This includes encrypted environment variables.
+	* Signing credentials – Stored signing credentials will be cloned along with passwords for the credentials.
+	* Build scripts – Build script settings will be cloned and applied. If the build script no longer exists in the repo, the build may fail.
+	* Distribution settings – Distribution to users, group, or store destinations will be cloned.
```

Comment:
Not sure about this. Usually we only distribute the master builds to app store etc. We should be able to select which configurations we want to clone.

Reply:
@SandroMachado from what I understand, everything will be copied, saved, and **then** you will have the ability to edit/remove any of the settings that you didn't want applied.

So in your situation, if you don't want the distribution settings to be copied, you would clone the configuration, save, then navigate to distribution settings on your cloned branch and turn them off.

Reply:
That's our thought here. You can always follow up an API call with another to adjust distribution settings and/or modify them through the portal UI directly.